packages/mds-jurisdiction-service/index.ts

```diff
@@ -111,6 +112,118 @@ const createJurisdiction = async (
+        .where({ jurisdiction_id })
+        .getOne()
+      if (entity) {
+        const current = AsJurisdiction()(entity)
+        if (current) {
+          const timestamp = updates.timestamp ?? Date.now()
+          if ((updates.jurisdiction_id ?? jurisdiction_id) !== current.jurisdiction_id) {
+            return Failure(new ValidationError('Invalid jurisdiction_id for update'))
+          }
+          if (timestamp <= current.timestamp) {
+            return Failure(new ValidationError('Invalid timestamp for update'))
+          }
+          const { id, ...update } = entity
+          const {
+            raw: [updated]
+          }: UpdateReturning<JurisdictionEntity> = await connection
+            .getRepository(JurisdictionEntity)
+            .createQueryBuilder()
+            .update()
+            .set({
+              ...update,
+              agency_key: updates.agency_key ?? current.agency_key,
+              versions:
+                (updates.agency_name ?? current.agency_name) !== current.agency_name ||
+                (updates.geography_id ?? current.geography_id) !== current.geography_id
+                  ? [
+                      {
+                        agency_name: updates.agency_name ?? current.agency_name,
+                        geography_id: updates.geography_id ?? current.geography_id,
+                        timestamp
+                      },
+                      ...entity.versions
+                    ].sort((a, b) => b.timestamp - a.timestamp)
+                  : entity.versions
+            })
+            .where('jurisdiction_id = :jurisdiction_id', { jurisdiction_id })
+            .returning('*')
+            .execute()
+          const jurisdiction = AsJurisdiction(timestamp)(updated)
+          return jurisdiction ? Success(jurisdiction) : Failure(new ServerError('Unexpected error during update'))
+        }
+      }
+      return Failure(new NotFoundError('Jurisdiction Not Found', { jurisdiction_id }))
+    } catch (error) /* istanbul ignore next */ {
+      await logger.error(error.message)
+      return Failure(error)
+    }
+  } catch (error) /* istanbul ignore next */ {
+    await logger.error(error.message)
+    return Failure(error instanceof ServerError ? error : new ServerError(error))
+  }
+}
+
+const deleteJurisdiction = async (
+  jurisdiction_id: UUID
+): Promise<JurisdictionServiceResult<Pick<Jurisdiction, 'jurisdiction_id'>, NotFoundError>> => {
+  try {
+    const connection = await manager.getReadWriteConnection()
+    try {
+      const entity = await connection
+        .getRepository(JurisdictionEntity)
+        .createQueryBuilder()
+        .where({ jurisdiction_id })
+        .getOne()
+      if (entity) {
+        const current = AsJurisdiction()(entity)
+        if (current) {
+          const { id, ...update } = entity
+          await connection
+            .getRepository(JurisdictionEntity)
+            .createQueryBuilder()
+            .update()
```

Comment:
Why is this `.update` and not `.delete` if this is meant to delete a Jurisdiction?

Reply:
Using `.update` to _soft_ delete a jurisdiction.

Reply:
Never mind, I read the spec.

Reply:
Adding a comment to clarify the behavior.